src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "math/integerhistogram1d.h"
+#include "base/lineparser.h"
+#include "base/messenger.h"
+#include "base/sysfunc.h"
+#include "math/mathfunc.h"
+#include "templates/algorithms.h"
+#include <cassert>
+
+IntegerHistogram1D::IntegerHistogram1D()
+{
+    accumulatedData_.addErrors();
+
+    clear();
+}
+
+IntegerHistogram1D::IntegerHistogram1D(const IntegerHistogram1D &source) { (*this) = source; }
+
+// Clear Data
+void IntegerHistogram1D::clear()
+{
+    nBinned_ = 0;
+    nMissed_ = 0;
+    bins_.clear();
+}
+
+/*
+ * Data
+ */
+
+// Update accumulated data
+void IntegerHistogram1D::updateAccumulatedData()
+{
+    accumulatedData_ = createDisplayData();
+    auto minBin = DissolveMath::nInt(accumulatedData_.xAxis().front());
+    // Poke bin values and errors into array
+    for (const auto &[key, value] : bins_)
+    {
+        auto n = key - minBin;
+        accumulatedData_.value(n) = value.second.value();
+        accumulatedData_.error(n) = value.second.stDev();
+    }
+}
+
+// Create dsiplay data
+const Data1D IntegerHistogram1D::createDisplayData()
+{
+    Data1D data;
+    // Get limiting key values
+    std::optional<int> expectedMinimum = bins_.empty() ? std::nullopt : std::optional<int>(bins_.begin()->first);
+    if (minimum_)
+        expectedMinimum = expectedMinimum ? std::min(*minimum_, *expectedMinimum) : *minimum_;
+
+    std::optional<int> expectedMaximum = bins_.empty() ? std::nullopt : std::optional<int>(std::prev(bins_.end())->first);
+    if (maximum_)
+        expectedMaximum = expectedMaximum ? std::max(*maximum_, *expectedMaximum) : *maximum_;
+
+    // Need to check on null values - set to zeroes if both null, or if one is null set it to match the other
+    if (!expectedMinimum && !expectedMaximum)
+    {
+        expectedMinimum = 0;
+        expectedMaximum = 0;
+    }
+    else if (!expectedMinimum)
+        expectedMinimum = expectedMaximum;
+    else if (!expectedMaximum)
+        expectedMaximum = expectedMinimum;
+
+    auto expectedNBins = (*expectedMaximum - *expectedMinimum) + 1;
+
+    // Set up data
+    data.initialise(expectedNBins, true);
+    auto x = *expectedMinimum;
+    for (auto n = 0; n < expectedNBins; n++)
```

Comment:
Being super-picky, but a prefix increment (`++n`) requires less CPU operations than a postfix increment (`n++`), and either could be used here.
```suggestion
    for (auto n = 0; n < expectedNBins; ++n)
```